### PR TITLE
Add `bevy_core::Name` doc examples

### PR DESCRIPTION
**Pull request by TimJentzsch**
# Objective

- Make it clear how `Name` and its functions are used.
- Add unit tests to make sure that everything works as expected.

## Solution

- Add examples to `Name`'s docs, which serve both as documentation and as tests.

### CODE REVIEW

**Comment by TimJentzsch** [comment]
# Preview

- `Name`:
    Name docs

- `Name::new()`:
    Name::new() docs

- `Name::set()`:
    Name::set() docs

- `Name::mutate()`:
    Name::mutate() docs

- `Name::as_str()`:
    Name::as_str() docs

**Comment by mockersf** [comment]
I would rather the docs don't show the usage of `as_str()` for a comparison with another string, one of the main point of `Name` is to have a faster `PartialEq` implementation that is using its hash

File: crates/bevy_core/src/name.rs [hunk 1]
```diff
@@ -8,12 +8,51 @@ use std::{
     ops::Deref,
 };
 
-/// Component used to identify an entity. Stores a hash for faster comparisons
+/// Component used to identify an entity. Stores a hash for faster comparisons.
```
Comment (alice-i-cecile) [inline crates/bevy_core/src/name.rs:11]: Line break after the first line of docs please :)

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Two larger requests:

1. Discuss how `Name` is used with scenes, and the gltf file format.
2. Demonstrate how to look up an entity by its name instead of showing `query_system`. For now, this is just a linear time search, but it's useful to show this still (to demonstrate that nothing better exists).

**Comment by TimJentzsch** [comment]
@alice-i-cecile 

> Discuss how Name is used with scenes, and the gltf file format.

Do you have any resources for me to look up here? I'm not really familiar with Bevy's scenes or the gltf format yet.

> Demonstrate how to look up an entity by its name instead of showing query_system. For now, this is just a linear time search, but it's useful to show this still (to demonstrate that nothing better exists).

I changed it to something similar as @mockersf suggested, please let me know if that works :)

**Comment by alice-i-cecile** [comment]
> Do you have any resources for me to look up here? I'm not really familiar with Bevy's scenes or the gltf format yet.

Ha, if we had good docs I wouldn't be asking you to write them! However, [here's](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#indices-and-names) a writeup of how they're used in `gltF`. When `gltF` files are loaded into Bevy as a scene, each gltF entity becomes one Bevy entity, and that property is translated into the `Name` component for the corresponding entity. You can see this in the `load_gltf` and associated `load_node` functions.

These are also apparently used in the `EntityPath` struct for animation, which is probably worth calling out as part of the section on gltF.

> I changed it to something similar as `@mockersf` suggested, please let me know if that works :)

I'm happy with those changes.

**Comment by mockersf** [comment]
Inside Bevy, names are very used for animations. You can see it in https://github.com/bevyengine/bevy/blob/main/examples/animation/animated_transform.rs where an animation is built in code. For glTF, every node is given a name, either the one from glTF file or a generated one so that it can be targeted by an animation.

**Comment by TimJentzsch** [comment]
I still don't really know how scenes/gltf work and currently don't have enough time to study them in detail.

I would propose to either factor that part out in a new issue or to put an Adopt Me tag on this PR :)

**Pull request closed without merging**